src/layouts/MainLayout.js

```diff
@@ -46,15 +46,18 @@ const MainLayout = ({ children }) => {
     ? null
     : `${siteMetadata.repository}/blob/main/${fileRelativePath}`;
 
-  useEffect(() => {
-    const consentValue = Cookies.get(gdprConsentCookieName) === 'true';
-    consentValue && setCookieConsent(true);
-  }, []);
-
   useEffect(() => {
     setIsMobileNavOpen(false);
   }, [location.pathname]);
 
+  useEffect(() => {
+    if (cookieConsent && previousCookieConsent === false) {
+      window.trackGoogleAnalytics();
+    }
+  }, [cookieConsent, previousCookieConsent]);
+
+  const previousCookieConsent = usePrevious(cookieConsent);
```

Comment:
I _think_ this may have trouble building since you're declaring this variable after you're using it on line 57. Could you move this above the previous `useEffect`?

Reply:
yes!